src/rgw/rgw_sal.cc

```diff
@@ -188,15 +186,24 @@ rgw::sal::Store* StoreManager::init_storage_provider(const DoutPrefixProvider* d
       ldpp_dout(dpp, 20) << "User display name = " << suser->get_info().display_name << dendl;
       ldpp_dout(dpp, 20) << "User email = " << suser->get_info().user_email << dendl;
     }
-
-    return store;
   }
 #endif
 
-  return nullptr;
+  if (filter.compare("base") == 0) {
+    rgw::sal::Store* next = store;
+    store = newBaseFilter(next);
+
+    if (store->initialize(cct, dpp) < 0) {
+      delete store;
+      delete next;
+      return nullptr;
+    }
+  }
+
+  return store;
 }
 
-rgw::sal::Store* StoreManager::init_raw_storage_provider(const DoutPrefixProvider* dpp, CephContext* cct, const std::string svc)
+rgw::sal::Store* StoreManager::init_raw_storage_provider(const DoutPrefixProvider* dpp, CephContext* cct, const std::string svc, const std::string filter)
```

Comment:
Isn't filter layer needed for raw_storage too? Maybe for some of the admin ops (like stats etc)

Reply:
Good catch